Minesweeper.py

```diff
@@ -43,68 +28,86 @@
                 print(board[y][x], end=" ")
         print()
 
-    # Print the game stats
+def print_stats():
     print("Moves:", moves)
     print("Flags:", flags)
     print("Mines found:", mines_found)
 
-    # Get the player's input
-    action = input("Enter your action (move, flag, or unflag): ")
-    direction = input("Enter the direction (up, down, left, or right): ")
-
-    # Update the player's position
+def update_player_pos(direction):
+    x, y = player_pos
     if direction == "up":
-        player_y -= 1
+        y -= 1
     elif direction == "down":
-        player_y += 1
+        y += 1
     elif direction == "left":
-        player_x -= 1
+        x -= 1
     elif direction == "right":
-        player_x += 1
-
-    # Check if the player has hit a mine
-    if board[player_y][player_x] == -1:
-        game_over = True
-        print("Game over! You hit a mine.")
-    else:
-        moves += 1
-
-    # Check if the player has won
-    if moves == BOARD_SIZE * BOARD_SIZE - NUM_MINES:
-        game_over = True
-        print("Congratulations! You found all the mines.")
-
-    # Place a mine
-    if action == "move":
-        # Do nothing
-        pass
-    elif action == "flag":
-        if board[player_y][player_x] == 0:
-            board[player_y][player_x] = "F"
-            flags += 1
-    elif action == "unflag":
-        if board[player_y][player_x] == "F":
-            board[player_y][player_x] = 0
-            flags -= 1
-
-# Place the mines randomly
-mine_count = 0
-while mine_count < NUM_MINES:
-    x = random.randint(0, BOARD_SIZE - 1)
-    y = random.randint(0, BOARD_SIZE - 1)
-    if board[y][x] == 0:
-        board[y][x] = -1
-        mine_count += 1
-
-# Calculate the number of mines around each cell
-for y in range(BOARD_SIZE):
-    for x in range(BOARD_SIZE):
-        if board[y][x] != -1:
-            count = 0
-            for dy in [-1, 0, 1]:
-                for dx in [-1, 0, 1]:
-                    nx = x + dx
-                    ny = y + dy
-                    if 0 <= nx < BOARD_SIZE and 0 <= ny < BOARD_SIZE and board[ny][nx] == -1:
-                        count += 1
-            board[y][x] = count
\ No newline at end of file
+        x += 1
```

Comment:
The `update_player_pos` function lacks appropriate bounds checking after updating the player's position. This could result in an index error if the player moves out of the board's limits. ```diff -    if direction == "up": -        y -= 1 -    elif direction == "down": -        y += 1 -    elif direction == "left": -        x -= 1 -    elif direction == "right": -        x += 1 +    if direction == "up" and y > 0: +        y -= 1 +    elif direction == "down" and y < BOARD_SIZE - 1: +        y += 1 +    elif direction == "left" and x > 0: +        x -= 1 +    elif direction == "right" and x < BOARD_SIZE - 1: +        x += 1 ``` ---